staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -174,7 +174,11 @@ func (a customResourceStrategy) Validate(ctx context.Context, obj runtime.Object
 
 		// validate x-kubernetes-validations rules
 		if celValidator, ok := a.celValidators[v]; ok {
-			errs = append(errs, celValidator.Validate(nil, a.structuralSchemas[v], u.Object)...)
+			err, remainingBudget := celValidator.Validate(nil, a.structuralSchemas[v], u.Object, cel.RuntimeCELCostBudget)
+			errs = append(errs, err...)
+			if remainingBudget < 0 {
+				return errs
+			}
```

Comment:
I think this short-circuit should be dropped, we're not skipping any further cel evaluation by doing this